src/providers/twitch/TwitchChannel.cpp

```diff
@@ -927,6 +927,69 @@ void TwitchChannel::refreshCheerEmotes()
         .execute();
 }
 
+void TwitchChannel::createClip()
+{
+    if (!this->isLive())
+    {
+        this->addMessage(makeSystemMessage(
+            "Cannot create clip while the channel is offline!"));
+        return;
+    }
+
+    this->addMessage(makeSystemMessage("Creating clip..."));
+
+    getHelix()->createClip(
+        this->roomId(),
+        [this](const HelixClip &clip) {
+            MessageBuilder builder;
+            builder.message().flags.set(MessageFlag::System);
+
+            builder.emplace<TimestampElement>();
+            // text
+            builder.emplace<TextElement>("Clip created!",
+                                         MessageElementFlag::Text,
+                                         MessageColor::System);
+            // clip link
+            builder
+                .emplace<TextElement>("Copy link to clipboard",
+                                      MessageElementFlag::Text,
+                                      MessageColor::Link)
+                ->setLink(Link(Link::CopyToClipboard,
+                               "https://clips.twitch.tv/" + clip.id));
```

Comment:
I would prefer the `"https://clips.twitch.tv/"` to be in a constant or something rather than a magic string

Reply:
I just assumed that this was going to be unchanged on Twitch's side so I allowed myself to "hardcode" this value in the said magic string to make things more simple. Though, I am a fan of the approach you've suggested and I will move that into a constant based on `QString` :+1: thanks for pointing this out